src/tests/end-to-end/common/page-controllers/page.ts

```diff
@@ -92,7 +98,16 @@ export class Page {
     }
 
     public async evaluate(fn: Puppeteer.EvaluateFn, ...args: any[]): Promise<any> {
-        return await this.screenshotOnError(async () => await this.underlyingPage.evaluate(fn, ...args));
+        const timeout = DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS;
+        return await this.screenshotOnError(
```

Comment:
is this `return await` redundant here? will `return this.screenshotOnError(` be sufficient?

Reply:
The return is required here because the function propogates the returned value of `fn`, similar to the original Puppeteer `evaluate` method (note the function return type is `Promise<any>`, not `Promise<void>`)

Reply:
Sorry, I misinterpreted the original comment; I see now you were asking about removing the `await`, not the `return`.

I think https://github.com/standard/standard/issues/1442 has a very good overview of why I think it is better to include the await as a general rule

Reply:
After readying the v8 article I agree!